TACC/Frontera-Portal
# Task/FP-694 metrics

## Overview: ##

Add logging statements to meet metric-collecting requirements

See JIRA ticket for more info.

## PR Status: ##

* [x] Work in Progress.  (Read but would like to add some more unit tests. See Todo)

## Related Jira tickets: ##

* [FP-694](https://jira.tacc.utexas.edu/browse/FP-694)

## Summary of Changes: ##
* Refactor existing "metric" logging
* Add logging for job submission
* Add logging for tapis file operations
* Add unit test to ensure job submission is logged for metrics
* Add unit test to ensure tapis file operation (in `apps/datafiles`) are logged for metric

## Testing Steps: ##
1. Run tests
2. Submit job and see logging statement in console
2. Perform some file operations (like listing or delete or move...) and see logging statement in console

## Notes: ##
I want to add unit tests to just make sure that any METRIC statements are covered.
TODO:
- [x] unit tests: port existing cep V1 `server/portal/apps/auth` unit tests
- [X] unit tests: add test to ensure Job create
- [ ] Add unit test to ensure tapis file operation (in `apps/datafiles`) are logged for metric
- [ ] remove extra from base.py as not used in our logging formatter
